addons/hr_expense/models/hr_expense.py

```diff
@@ -394,24 +395,31 @@ def _create_sheet_from_expenses(self):
             raise UserError(_("You can not create report without category."))
 
         todo = self.filtered(lambda x: x.payment_mode=='own_account') or self.filtered(lambda x: x.payment_mode=='company_account')
-        sheet = self.env['hr.expense.sheet'].create({
-            'company_id': self.company_id.id,
-            'employee_id': self[0].employee_id.id,
-            'name': todo[0].name if len(todo) == 1 else _('Expense Report'),
-            'expense_line_ids': [(6, 0, todo.ids)]
-        })
-        return sheet
+        if len(todo) == 1:
+            expense_name = todo.name
+        else:
+            dates = todo.mapped('date')
+            expense_name = format_date(self.env, min(dates)) if max(dates) == min(dates) else "%s - %s" % (format_date(self.env, min(dates)), format_date(self.env, max(dates)))
```

Comment:
You could save `min(dates)` and `max(dates)`, as they are used twice

Reply:
@kba-odoo , Done the changes. Thanks